Check the existence of the index pattern of vulnerabilities states in Vulnerabilities Detection tabs

Desvelao opened the pull request:
### Description
This pull request adds a check about the existence of the `wazuh-states-vulnerabilities` index pattern to the Vulnerabilities Detection tabs (Dashboard, Inventory, Events) and if the module is enabled (only Dashboard and Inventory).

Changes:
- Check the existence of the index pattern: check if the index pattern exists, if not, then check if there are fields for the index pattern (some index should exist), if so, then creates silently the index pattern.
> :warning: The index pattern uses the `vulnerabilities.pattern` setting. Title and ID of the index pattern should be the value of `vulnerabilities.pattern`. This behavior was not changed.

- Check if the vulnerabilities module is enabled:
  - Check if the cluster is enabled or not.
    - Cluster enabled: get the list of nodes and check if at leat someone of them it has the module enabled. If so, then it is considered the module is enabled.
    > :warning: The use case of some nodes has not the module enabled is not managed
    - Cluster disabled: check if the module is enabled in the manager.
    
 - Removed the check of the existence of related index on Dashboard and Inventory in favor of the check of the index pattern.

### Issues Resolved
#6394

### Evidence
No index pattern and it could not be created (matching index doesn't exist):
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/34042064/afdec8e0-f770-415e-9b4b-9b015d5a5dec)
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/34042064/a94e4dbc-1eaa-48a7-9f50-dce1cb0a7f2b)
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/34042064/734dedc4-f2a1-4f17-b64c-28d65e20e11e)

- Callout related to the module is not enabled
  - Dashboard
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/34042064/349ace5f-c71f-4dc4-9d1e-b5158be8890a)

  - Inventory
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/34042064/fed6a85a-d5e7-4eff-9f10-bca8c23f1702)

- No callout displayed if the module is enabled
- Dashboard
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/34042064/f6a347f4-6813-49ab-84cc-12a4836f093a)
- Inventory
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/34042064/60799ce3-32f0-4d2b-841b-bd9a57283b73)

- Checking data source loading effect
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/34042064/8609c95b-c290-48bf-bcac-6072abae7b70)

### Test
Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and index created, go to Vulnerabilities Detection > Dashboard, the index pattern must be created, redirect to the Dashboard tab and display the view | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and index created, go to Vulnerabilities Detection > Inventory, the index pattern must be created, redirect to the Dashboard tab and display the view | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and index created, go to Vulnerabilities Detection > Events, the index pattern must be created, redirect to the Dashboard tab and display the view | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is disabled with vulnerabilities module enabled , go to Vulnerabilities Detection > Dashboard, must not display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is disabled with vulnerabilities module disabled , go to Vulnerabilities Detection > Inventory, must display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is enabled with vulnerabilities module enabled in some of the nodes, go to Vulnerabilities Detection > Dashboard, must not display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is enabled with vulnerabilities module disabled in all the nodes, go to Vulnerabilities Detection > Dashboard, must display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is disabled with vulnerabilities module enabled , go to Vulnerabilities Detection > Inventory, must not display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is disabled with vulnerabilities module disabled , go to Vulnerabilities Detection > Inventory, must display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is enabled with vulnerabilities module enabled in some of the nodes, go to Vulnerabilities Detection > Inventory, must not display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is enabled with vulnerabilities module disabled in all the nodes, go to Vulnerabilities Detection > Inventory, must display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and index created, go to Vulnerabilities Detection > Dashboard, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and index created, go to Vulnerabilities Detection > Inventory, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and index created, go to Vulnerabilities Detection > Events, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is disabled with vulnerabilities module enabled , go to Vulnerabilities Detection > Dashboard, must not display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is disabled with vulnerabilities module disabled , go to Vulnerabilities Detection > Inventory, must display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is enabled with vulnerabilities module enabled in some of the nodes, go to Vulnerabilities Detection > Dashboard, must not display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is enabled with vulnerabilities module disabled in all the nodes, go to Vulnerabilities Detection > Dashboard, must display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is disabled with vulnerabilities module enabled , go to Vulnerabilities Detection > Inventory, must not display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is disabled with vulnerabilities module disabled , go to Vulnerabilities Detection > Inventory, must display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is enabled with vulnerabilities module enabled in some of the nodes, go to Vulnerabilities Detection > Inventory, must not display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is enabled with vulnerabilities module disabled in all the nodes, go to Vulnerabilities Detection > Inventory, must display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

lucianogorza commented:
## CR 🟢 
## Test ⚫ 
| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the view should be visible | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the view should be visible | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the view should be visible | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and index created, go to Vulnerabilities Detection > Dashboard, must display the view and the index pattern must have been created | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and index created, go to Vulnerabilities Detection > Inventory, must display the view and the index pattern must have been created | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and index created, go to Vulnerabilities Detection > Events, must display the view and the index pattern must have been created | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is disabled with vulnerabilities module enabled , go to Vulnerabilities Detection > Dashboard, must not display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is disabled with vulnerabilities module disabled , go to Vulnerabilities Detection > Dashboard, must display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is enabled with vulnerabilities module enabled in some of the nodes, go to Vulnerabilities Detection > Dashboard, must not display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is enabled with vulnerabilities module disabled in all the nodes, go to Vulnerabilities Detection > Dashboard, must display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is disabled with vulnerabilities module enabled , go to Vulnerabilities Detection > Inventory, must not display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is disabled with vulnerabilities module disabled , go to Vulnerabilities Detection > Inventory, must display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is enabled with vulnerabilities module enabled in some of the nodes, go to Vulnerabilities Detection > Inventory, must not display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is enabled with vulnerabilities module disabled in all the nodes, go to Vulnerabilities Detection > Inventory, must display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the view should be visible</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the view should be visible</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the view should be visible</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and index created, go to Vulnerabilities Detection > Dashboard, must display the view and the index pattern must have been created</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and index created, go to Vulnerabilities Detection > Inventory, must display the view and the index pattern must have been created</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and index created, go to Vulnerabilities Detection > Events, must display the view and the index pattern must have been created</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is disabled with vulnerabilities module enabled , go to Vulnerabilities Detection > Dashboard, must not display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is disabled with vulnerabilities module disabled , go to Vulnerabilities Detection > Dashboard, must display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is enabled with vulnerabilities module enabled in some of the nodes, go to Vulnerabilities Detection > Dashboard, must not display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is enabled with vulnerabilities module disabled in all the nodes, go to Vulnerabilities Detection > Dashboard, must display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is disabled with vulnerabilities module enabled , go to Vulnerabilities Detection > Inventory, must not display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is disabled with vulnerabilities module disabled , go to Vulnerabilities Detection > Inventory, must display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is enabled with vulnerabilities module enabled in some of the nodes, go to Vulnerabilities Detection > Inventory, must not display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is enabled with vulnerabilities module disabled in all the nodes, go to Vulnerabilities Detection > Inventory, must display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

lucianogorza commented on plugins/main/public/components/overview/vulnerabilities/common/components/check-module-enabled.tsx:
```diff
@@ -0,0 +1,115 @@
+        configure following the{' '}
+        <EuiLink
+          href={webDocumentationLink(
+            'user-manual/capabilities/vulnerability-detector/configuring-scans.html#configuring-vulnerability-detector.html',
```
```suggestion
            'user-manual/capabilities/vulnerability-detection/configuring-scans.html#configuring-vulnerability-detector.html',
```

Desvelao replied:
I fixed the URL. There was another error in the section part too: `#configuring-vulnerability-detector.html` -> `#configuring-vulnerability-detection`

Desvelao commented:
>  With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the view should be visible

@lucianogorza , I added a fix to that problem.

Due to the current flow (coming from previous work) that sets the filters that are placed in Dashboard or Inventory tabs and that they could not have been defined due to the usage of the HOC, if the index pattern is created, we redirect to the root of the Vulnerability Detection that redirects to the `Dashboard` tab. If the dashboard is visible, and navigating to the `Events` tab, the logic to set the filters should be run, and the expected implicit filter should be visible.

This is only a workaround to the current flow to set the filters. It should be redone in the future moving the responsibility to set the filters to the view that will be used instead.

yenienserrano commented:
## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and index created, go to Vulnerabilities Detection > Dashboard, the index pattern must be created, redirect to the Dashboard tab and display the view | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and index created, go to Vulnerabilities Detection > Inventory, the index pattern must be created, redirect to the Dashboard tab and display the view | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and index created, go to Vulnerabilities Detection > Events, the index pattern must be created, redirect to the Dashboard tab and display the view | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is disabled with vulnerabilities module enabled , go to Vulnerabilities Detection > Dashboard, must not display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is disabled with vulnerabilities module disabled , go to Vulnerabilities Detection > Inventory, must display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is enabled with vulnerabilities module enabled in some of the nodes, go to Vulnerabilities Detection > Dashboard, must not display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is enabled with vulnerabilities module disabled in all the nodes, go to Vulnerabilities Detection > Dashboard, must display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is disabled with vulnerabilities module enabled , go to Vulnerabilities Detection > Inventory, must not display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is disabled with vulnerabilities module disabled , go to Vulnerabilities Detection > Inventory, must display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is enabled with vulnerabilities module enabled in some of the nodes, go to Vulnerabilities Detection > Inventory, must not display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |
| With index pattern and index and cluster mode is enabled with vulnerabilities module disabled in all the nodes, go to Vulnerabilities Detection > Inventory, must display a callout about the module is disabled | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:green_circle: With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem</summary>

Chrome - :green_circle:

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/63758389/7a26e9a0-91c1-45c2-86b0-840c4390c75a)

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem</summary>

Chrome - :green_circle:

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/63758389/48206195-04c9-4003-9ff9-bee4e900c8ad)

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem</summary>

Chrome - :green_circle:

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/63758389/1998cfc4-db94-43f3-9c13-0c8b6b4182a6)

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again</summary>

Chrome - :green_circle:

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/63758389/ba326995-6cf7-4b7b-ab5d-9a9be093d6d9)

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again</summary>

Chrome - :green_circle:

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/63758389/9eb627b7-a811-4a8c-b347-a6766879ccb8)

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem, click on the Refresh button, and the prompt must appear again</summary>

Chrome - :green_circle:

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/63758389/575814d6-578e-442f-a92a-52e77c75cf9e)

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and index created, go to Vulnerabilities Detection > Dashboard, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and index created, go to Vulnerabilities Detection > Inventory, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no index pattern and index created, go to Vulnerabilities Detection > Events, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is disabled with vulnerabilities module enabled , go to Vulnerabilities Detection > Dashboard, must not display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is disabled with vulnerabilities module disabled , go to Vulnerabilities Detection > Inventory, must display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is enabled with vulnerabilities module enabled in some of the nodes, go to Vulnerabilities Detection > Dashboard, must not display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is enabled with vulnerabilities module disabled in all the nodes, go to Vulnerabilities Detection > Dashboard, must display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is disabled with vulnerabilities module enabled , go to Vulnerabilities Detection > Inventory, must not display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is disabled with vulnerabilities module disabled , go to Vulnerabilities Detection > Inventory, must display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is enabled with vulnerabilities module enabled in some of the nodes, go to Vulnerabilities Detection > Inventory, must not display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With index pattern and index and cluster mode is enabled with vulnerabilities module disabled in all the nodes, go to Vulnerabilities Detection > Inventory, must display a callout about the module is disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

lucianogorza commented:
## CR 🟢 
## Test 🟢 

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view | 🟢 | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view | 🟢 | :black_circle: | :black_circle: |
| With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view |🟢 | :black_circle: | :black_circle: |
| With no index pattern and index created, go to Vulnerabilities Detection > Dashboard, the index pattern must be created, redirect to the Dashboard tab and display the view | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and index created, go to Vulnerabilities Detection > Inventory, the index pattern must be created, redirect to the Dashboard tab and display the view | :black_circle: | :black_circle: | :black_circle: |
| With no index pattern and index created, go to Vulnerabilities Detection > Events, the index pattern must be created, redirect to the Dashboard tab and display the view | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>🟢 With no index pattern and no index, go to Vulnerabilities Detection > Dashboard, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - 🟢

[dahsboard-refresh.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/103193307/75d3d343-f27c-4b85-ad37-a233931dbb02)

</details>

<details>
<summary>🟢 With no index pattern and no index, go to Vulnerabilities Detection > Inventory, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - 🟢

[inventory-refresh.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/103193307/bf74c326-f0df-448b-b8be-29905f047c32)

</details>

<details>
<summary>🟢 With no index pattern and no index, go to Vulnerabilities Detection > Events, a prompt must indicate the module could be disabled or there is a problem, ensure the index is created and click on the Refresh button, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - 🟢 

[events-refresh.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/103193307/60742ac9-ba65-4955-84d5-1cc4d815683e)

</details>

<details>
<summary>:black_circle: With no index pattern and index created, go to Vulnerabilities Detection > Dashboard, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - :black_circle:


</details>

<details>
<summary>:black_circle: With no index pattern and index created, go to Vulnerabilities Detection > Inventory, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - :black_circle:


</details>

<details>
<summary>:black_circle: With no index pattern and index created, go to Vulnerabilities Detection > Events, the index pattern must be created, redirect to the Dashboard tab and display the view</summary>

Chrome - :black_circle:


</details>

lucianogorza reviewed (approved):
LGTM

Desvelao commented:
Added a loading effect while checking the existence of the `wazuh-states-vulnerabilities`:
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/34042064/8609c95b-c290-48bf-bcac-6072abae7b70)

Fixed a problem found by @yenienserrano related to repeated requests on the Inventory tab when checking if the module is enabled or not.

lucianogorza reviewed (approved):
LGTM

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/103193307/88a0a81b-8a53-4d84-b194-4002a2abb900)

yenienserrano reviewed (approved):
CR: ✅ 
Test: ✅ 

https://github.com/wazuh/wazuh-dashboard-plugins/assets/63758389/305abc35-ab98-4d97-b80a-f3d2b86291aa
